### ISSUE TITLE
Research Using Third-Party Command-Line Option Parsing Library

### ISSUE THREAD

**Issue by ceccon-t** [ceccon-t/JenAI#3]
## Description

JenAI currently parses command-line options manually, which makes the code less legible and maintainable. To improve these aspects, it would be beneficial to leverage a well-established, third-party Java library for command-line option parsing. 

## Problem Statement

The current implementation of parsing command-line options within JenAI's codebase leads to several issues:
- **Code Duplication and Repetition**: The parsing logic duplicates several parts of low-level processing, making it harder to understand the flow and maintain consistency.
- **Error Handling Complexity**: Manual parsing can lead to more complex error handling, as the application needs to manage and report issues related to option validation and incorrect usage.
- **Limited Flexibility and Extensibility**: Hardcoding parsing logic restricts the ability to easily add or modify options in the future, potentially requiring significant changes to the application's core.

## Solution Proposal

To address these challenges, the suggestion is to explore and integrate a reputable, open-source Java library for command-line option parsing. Some popular options include:
- Apache Commons CLI
- JCommander
- Args4J

The chosen library should provide features such as:
- Easy option definition and validation
- Support for various option types (e.g., flags, values, subcommands)
- Built-in help and usage message generation
- Error handling and reporting mechanisms
- Compatibility with Java 21 and above (JenAI's current target version)

**Acceptance Criteria:**

1. The selected library is thoroughly tested in a local environment to ensure compatibility with JenAI's existing architecture and command-line interface.
2. All manual command-line option parsing logic is refactored to utilize the chosen library, promoting code consistency and readability.
3. Error handling and reporting mechanisms are updated to utilize the library's features, providing a more robust and user-friendly experience.
4. The migration to the new parsing library does not introduce any significant performance regressions or compatibility issues with existing use cases.

**Comment by ceccon-t** [comment]
A good starting point is Apache Commons CLI. 

Consider the following resources:
- [Official documentation](https://commons.apache.org/proper/commons-cli/)
- [Usage example](https://stackoverflow.com/questions/367706/how-do-i-parse-command-line-arguments-in-java)

**Comment by ceccon-t** [comment]
This is also a great opportunity to add a suite of tests for the configuration built from the arguments to the main function. Consider either making the parse method public, or testing through `main` itself (might have to create mock/spy of `CLISession` in that case).

**Comment by ceccon-t** [comment]
Another alternative to check out is JCommander.

Some resources:
- [Official documentation](https://jcommander.org/)
- [Tutorial with some examples](https://www.baeldung.com/jcommander-parsing-command-line-parameters)

**Comment by ceccon-t** [comment]
After checking out both Apache Commons CLI and JCommander, I chose to pick JCommander from version 1.7.1 onwards.

While both libraries worked fine for parsing the options, JCommander has a more modern approach (using a class to represent the possible options and annotations to configure them) and offered built-in type validation - maybe Apache Commons CLI also can be configured to validate types, but I did not find anything about it while reading the documentation and setting it up in the code.

Also, before checking both libraries I wrote unit tests for all of the current command line options. I tried to do this through the main method initially, but that quickly got quite messy, so instead I made the method that parses the options public and tested it directly. As a bonus, that allowed for changing several static properties into local variables, which I think ended up being a better design overall.
